### ISSUE TITLE
Issue #37: Large-Scale Screening - Validate 73% Coverage on 1000+ Molecules

### ISSUE THREAD

**Issue by kavanaghpatrick** [kavanaghpatrick/aristotle_biochemistry#37]
## 🎯 Objective

**Scale formal verification to 1000+ molecules** to validate coverage, generate ML training data, and guide future work.

## 📊 Current State

- **37 molecules** manually curated test set
- **27 proven safe** (73.0% coverage)
- **10 unprovable** with current methods

**Methods**:
1. Reachability (radius < 6.35 Å)
2. Volume exclusion (volume > 1810 Å³)
3. Pharmacophore (no aromatic/nitrogen or distance violation)

### Primary
1. **Validate 73% coverage holds at scale** (statistical significance)
2. **Generate 700+ proven-safe molecules** for ML training
3. **Characterize unprovable 27%** to guide future work (Issue #35)

### Secondary
4. **Demonstrate industry practicality** (can screen libraries)
5. **Identify edge cases** that might break our assumptions
6. **Publication-quality dataset** for Nature Comms / JCIM

### Data Source
- **ChEMBL API** (primary): hERG-tested molecules
  - Filter: IC50 > 10 µM or inactive (non-binders)
  - Target: 1000-2000 molecules
- **Backup**: DrugBank approved drugs if ChEMBL fails

### Proof Methods (Apply All)
1. **Reachability**: `radius < 6.35 Å`
2. **Volume**: `volume > 1810 Å³`
3. **Pharmacophore**: 
   - No aromatic rings
   - No basic nitrogen
   - Distance violation (all pairs outside [5, 7] Å)

### Output
```json
{
  "proven_safe": [
    {
      "chembl_id": "CHEMBL123",
      "smiles": "...",
      "proof": {
        "method": "reachability",
        "radius": 5.8,
        "reason": "Radius 5.8 Å < 6.35 Å"
      },
      "molecular_properties": {...}
    }
  ],
  "unprovable": [
    {
      "chembl_id": "CHEMBL456",
      "smiles": "...",
      "molecular_properties": {
        "radius": 7.5,
        "volume": 1500,
        "aromatic_count": 2,
        "nitrogen_count": 1
      }
    }
  ],
  "statistics": {
    "total": 1000,
    "proven_safe": 730,
    "coverage": 73.0
  }
}
```

### Phase 1: ChEMBL Data Collection (Day 1)
```python
# File: validation_suite/large_scale_screening.py

def query_chembl_herg_molecules(limit=2000):
    # Query ChEMBL API for hERG bioactivity
    # Filter for non-binders (IC50 > 10 µM)
    # Return list of {chembl_id, smiles, activity_data}
```

**Success criteria**: Load 1000-2000 molecules

### Phase 2: Apply Proof Methods (Day 1-2)
```python
def screen_molecule(smiles):
    # Try reachability
    if proof := check_reachability(mol):
        return proof
    
    # Try volume
    if proof := check_volume_exclusion(mol):
        return proof
    
    # Try pharmacophore
    if proof := check_pharmacophore(mol):
        return proof
    
    # Unprovable
    return None
```

**Success criteria**: Process all molecules, categorize results

### Phase 3: Statistical Analysis (Day 2)
```python
def analyze_results(results):
    # Coverage by method
    # Property distributions (radius, volume, MW)
    # Unprovable molecule characterization
    # Identify patterns in the "hard 27%"
```

### Phase 4: ML Export (Day 2)
```python
def export_for_ml(proven_safe):
    # Export to CSV/JSON for scikit-learn
    # Features: SMILES, fingerprints, descriptors
    # Label: proven_safe=True
```

### Must Have
- [ ] Load ≥1000 molecules from ChEMBL
- [ ] Apply all 3 proof methods automatically
- [ ] Achieve coverage ≥60% (validates methodology)
- [ ] Generate comprehensive results JSON
- [ ] 0% false negatives (no proven-safe binders)

### Should Have
- [ ] Coverage close to 73% (±5% validates our test set)
- [ ] Clear patterns in unprovable molecules
- [ ] Exportable ML training data
- [ ] Statistical analysis (confidence intervals)

### Nice to Have
- [ ] Coverage >75% (better than expected!)
- [ ] Identify new proof opportunities in unprovable set
- [ ] Visualization (coverage by method, property distributions)

### Scenario A: Coverage ~73% (Most Likely)
- ✅ Validates test set was representative
- ✅ 730+ proven-safe molecules for ML
- ✅ Publish results as-is
- 🔬 Analyze unprovable 27% → decide on Issue #35

### Scenario B: Coverage 80-90% (Optimistic)
- 🎉 Even better than expected!
- ✅ Test set was slightly pessimistic
- ✅ 800-900 proven-safe molecules
- 🔬 Remaining 10-20% very hard (algebraic methods?)

### Scenario C: Coverage 60-70% (Pessimistic)
- ⚠️ Test set was slightly optimistic
- ⚠️ Still 600-700 proven-safe molecules (valuable!)
- 🔬 30-40% unprovable → significant gap remains
- 🤔 Re-evaluate methodology

## 🔍 Analysis Questions to Answer

1. **Does 73% hold?** Coverage within ±5%?
2. **What characterizes the unprovable 27%?**
   - Size range (radius, volume)?
   - Flexibility (rotatable bonds)?
   - Structural scaffolds?
   - Pharmacophore patterns?
3. **Are there subgroups with >90% coverage?**
   - Small molecules?
   - Rigid scaffolds?
   - Simple pharmacophores?
4. **Should we pursue Issue #35 (Semi-Algebraic)?**
   - Do unprovables show algebraic patterns?
   - Would polynomial constraints help?

### Immediate (Week 1)
- **730+ proven-safe molecules** → ML training data
- **Statistical validation** → publication strength
- **Pattern analysis** → guide future work

### Short-term (Months)
- **Better hERG ML models** (industry value)
- **Publication** in Nature Comms / JCIM
- **Industry demo** (can screen libraries)

### Long-term (Years)
- **Hybrid formal+ML system** (production use)
- **Template for other targets** (generalization)
- **Regulatory acceptance** (pathway established)

### Code
- `validation_suite/large_scale_screening.py` - Main pipeline
- `validation_suite/chembl_query.py` - ChEMBL API wrapper
- `validation_suite/analysis.py` - Statistical analysis

### Data
- `large_scale_screening_results.json` - Full results
- `proven_safe_ml_training.csv` - ML-ready export
- `unprovable_analysis.json` - Characterization

### Reports
- `LARGE_SCALE_RESULTS.md` - Summary + statistics
- `UNPROVABLE_ANALYSIS.md` - Pattern analysis
- Visualizations (coverage plots, property distributions)

## ⏱️ Timeline

- **Day 1 AM**: Implement ChEMBL query + screening pipeline
- **Day 1 PM**: Run on first 100 molecules (validate)
- **Day 2 AM**: Full run on 1000 molecules (~2-3 hours)
- **Day 2 PM**: Analysis + report generation
- **Day 3**: Review, iterate, finalize

**Total**: 2-3 days

## 🔗 Dependencies

- **Enables**: Decision on Issue #35 (Semi-Algebraic)
- **Blocks**: Publication (need statistical validation)
- **Related**: Issue #30 (Positive Controls), Issues #19-21 (Documentation)

### Validation
- Cross-check random sample (10%) against experimental data
- Ensure 0% false positives (proven-safe actually non-binders)
- Statistical significance (p < 0.001 for coverage)

### Documentation
- Cite ChEMBL data source
- Document filtering criteria
- Explain any molecules excluded (invalid SMILES, etc.)
- Transparent about limitations

## 💡 Key Insights This Will Reveal

1. **Is 73% a universal limit** or test-set artifact?
2. **What makes molecules unprovable** with current methods?
3. **Is the methodology ready for industry** deployment?
4. **Should we invest in Issue #35** (Semi-Algebraic)?

---

**This is the critical validation step before publication!**
